File: youlexuan_parent/youlexuan_seckill_web/src/main/webapp/js/service/seckillGoodsService.js
//seckill_goods服务层
app.service('seckillGoodsService', function($http){

	//提交订单
	this.submitOrder=function(seckillId){
		return $http.get('seckillOrder/submitOrder.do?seckillId='+seckillId);
	}


//读取列表数据绑定到表单中
	this.findList=function(){
		return $http.get('seckillGoods/findList.do');
	}

	this.findOne=function(id){
		return $http.get('seckillGoods/findOneFromRedis.do?id='+id);
	}


/*******************************这是分割线，以上功能都是用到的**********************************/
	// 保存、修改
	this.save = function(entity) {
		var methodName = 'add'; 	// 方法名称
		if (entity.id != null) { 	// 如果有ID
			methodName = 'update'; 	// 则执行修改方法
		}
		return $http.post('../seckillGoods/' + methodName + '.do', entity);
	}

	// 查询单个实体
	this.findOne = function(id) {
		return $http.get('../seckillGoods/findOne.do?id=' + id);
	}

	// 批量删除
	this.dele = function(ids) {
		// 获取选中的复选框
		return $http.get('../seckillGoods/delete.do?ids=' + ids);
	}

	// 查询
	this.search = function(page, size, searchEntity) {
		// post提交，page、size属性和之前相同，将searchEntity提交至后台@RequestBody对应的属性
		return $http.post('../seckillGoods/search.do?page=' + page + '&size=' + size,
				searchEntity);
	}

});

File: youlexuan_parent/youlexuan_seckill_interface/src/main/java/com/lazy/seckill/service/SeckillGoodsService.java
package com.lazy.seckill.service;

import com.lazy.entity.PageResult;
import com.lazy.pojo.TbSeckillGoods;

import java.util.List;

/**
 * seckill_goods服务层接口
 * @author lazy
 *
 */
public interface SeckillGoodsService {

	/**
	 * 返回当前正在参与秒杀的商品
	 * @return
	 */
	public List<TbSeckillGoods> findList();

/*******************************这是分割线，以上功能都是用到的**********************************/
	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbSeckillGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbSeckillGoods seckillGoods);


	/**
	 * 修改
	 */
	public void update(TbSeckillGoods seckillGoods);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbSeckillGoods findOne(Long id);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbSeckillGoods seckillGoods, int pageNum, int pageSize);
	
}
